compiler/rustc_resolve/src/imports.rs

```diff
@@ -388,15 +439,21 @@ impl<'a, 'tcx> Resolver<'a, 'tcx> {
 
             let t = f(self, resolution);
 
-            if old_binding.is_none() && let Some(binding) = resolution.binding() {
+            update = is_update || old_binding.is_some();
```

Comment:
Isn't `old_binding` *always* `Some` when `is_update` is passed?

Reply:
Yes, in most cases, `﻿is_update` is originally set to ﻿false, and we require ﻿`old_binding.is_some()` to set ﻿update to ﻿true.

By the way, `﻿is_update` was introduced in ﻿`issue-113099-10.rs` to handle a specific scenario where we need to set ﻿update as ﻿true, even if `﻿old_binding` is `﻿None` due to a recursive update.